This gcc bug is fixed in the gcc-4.8 package in debian.

--- a/urcu/compiler.h
+++ b/urcu/compiler.h
@@ -117,14 +117,6 @@
 				+ __GNUC_MINOR__ * 100 \
 				+ __GNUC_PATCHLEVEL__)
 
-/*
- * http://gcc.gnu.org/bugzilla/show_bug.cgi?id=58854
- */
-# ifdef __ARMEL__
-#  if URCU_GCC_VERSION >= 40800 && URCU_GCC_VERSION <= 40802
-#   error Your gcc version produces clobbered frame accesses
-#  endif
-# endif
 #endif
 
 #endif /* _URCU_COMPILER_H */
